docs/source/guide/qse-3-options.md

```diff
@@ -0,0 +1,49 @@
+---
+jupytext:
+  text_representation:
+    extension: .myst
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.11.1
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+
+# What additional options are available in QSE?
+In addition to the necessary ingredients already discussed in [How do I use QSE?](qse-1-intro.md), there are a few additional options included in the implementation. 
+
+## Caching Pauli Strings to Expectation Values
+
+Specifically, in order to save runtime, the QSE implementation supports the use of a cache that maps pauli strings to their expectation values. This is taken as an additional parameter in the “execute_with_qse” function reproduced below. 
+
+```
+def execute_with_qse(
+    circuit: QPROGRAM,
+    executor: Union[Executor, Callable[[QPROGRAM], QuantumResult]],
+    check_operators: Sequence[PauliString],
+    code_hamiltonian: Observable,
+    observable: Observable,
+    pauli_string_to_expectation_cache: Dict[PauliString, complex] = {},
+) -> float:
+    """Function for the calculation of an observable from some circuit of
+        interest to be mitigated with Quantum Subspace Expansion
+    Args:
+        circuit: Quantum program to execute with error mitigation.
+        executor: Executes a circuit and returns a `QuantumResult`.
+        check_operators: List of check operators that define the
+        stabilizer code space.
+        code_hamiltonian: Hamiltonian of the code space.
+        observable: Observable to compute the mitigated expectation value of.
+        pauli_string_to_expectation_cache: Cache for expectation values of
+        Pauli strings used to compute the projector and the observable.
+    Returns:
+        The expectation value estimated with QSE.
+```
+The inclusion of the cache significantly speeds up the runtime and avoids the need for recomputation of already computed values. Furthermore, it is important to note that the cache gets modified in place. Specifically, the cache be reused between different executions of the program. 
```

Comment:
Let's use the `warning` directive to warn users that this dictionary gets modified in place. Something like this:
````md
```{warning}
warning message about in place modification
```
````
More info about directives can be found [here](https://myst-parser.readthedocs.io/en/latest/syntax/roles-and-directives.html).

Also, can you clarify the last sentence "Specifically, the cache be reused between different executions of the program."?

Reply:
Needs addressed.

Reply:
Added the warning message.

I'm not sure how to clarify the last sentence. I'm trying to say that the user can use the pass the same cache object to `execute_with_qse` for example if they are assuming the same noise model between executions.